edb/edgeql/parser/grammar/ddl.py

```diff
@@ -2865,22 +2871,44 @@ def reduce_CreateMigration(self, *kids):
             CREATE MIGRATION OptMigrationNameParentName
             OptCreateMigrationCommandsBlock
         """
+        message = None
+        generated_by = None
+        for f in kids[3].val.fields:
+            match f.name:
+                case 'message':
+                    message = f.value
+                case 'generated_by':
+                    generated_by = f.value
+
         self.val = qlast.CreateMigration(
             name=kids[2].val.name,
             parent=kids[2].val.parent,
             body=kids[3].val.body,
+            message=message,
```

Comment:
I think that if we populate the `commands` field here with `kids[3].val.fields`, the generic mechanisms for handling `SET field := ...` should pick up the commands and work without needing to have a `message` or `generated_by` field in `qlast.CreateMigration`, without needing to pick them out in the parser, and thus also without needing to handle them in `_cmd_from_ast`